diff --git a/node_modules/@auth/core/lib/routes/session.js b/node_modules/@auth/core/lib/routes/session.js
index e496d92..483c6e8 100644
--- a/node_modules/@auth/core/lib/routes/session.js
+++ b/node_modules/@auth/core/lib/routes/session.js
@@ -91,10 +91,7 @@ export async function session(params) {
             const sessionPayload = await callbacks.session({
                 // By default, only exposes a limited subset of information to the client
                 // as needed for presentation purposes (e.g. "you are logged in as...").
-                session: {
-                    user: { name: user.name, email: user.email, image: user.image },
-                    expires: session.expires.toISOString(),
-                },
+                session,
                 user,
                 newSession,
                 ...(isUpdate ? { trigger: "update" } : {}),
