diff --git a/dist/adapters/next-app-dir/nextAppDirCaller.js b/dist/adapters/next-app-dir/nextAppDirCaller.js
index 9b54e271c012f2c72a8df4dbf869641e3cbbd474..3a92b32372ecf2263cce714667d0d215bbc6e541 100644
--- a/dist/adapters/next-app-dir/nextAppDirCaller.js
+++ b/dist/adapters/next-app-dir/nextAppDirCaller.js
@@ -4,7 +4,7 @@ var TRPCError = require('../../unstable-core-do-not-import/error/TRPCError.js');
 require('../../unstable-core-do-not-import/rootConfig.js');
 var formDataToObject = require('./formDataToObject.js');
 var redirect = require('./redirect.js');
-var rethrowNextErrors = require('./rethrowNextErrors.js');
+var nextNavigation = require('next/navigation.js');
 
 /**
  * Create a caller that works with Next.js React Server Components & Server Actions
@@ -34,7 +34,7 @@ var rethrowNextErrors = require('./rethrowNextErrors.js');
                 path,
                 type: opts._def.type
             });
-            rethrowNextErrors.rethrowNextErrors(error);
+            nextNavigation.unstable_rethrow(error);
             throw error;
         };
         switch(opts._def.type){
diff --git a/dist/adapters/next-app-dir/nextAppDirCaller.mjs b/dist/adapters/next-app-dir/nextAppDirCaller.mjs
index afea53905536e4344ac4dbd0d0debd5710ad7cc9..64f284f145089d1835cd51914daa8d2aae3a1c48 100644
--- a/dist/adapters/next-app-dir/nextAppDirCaller.mjs
+++ b/dist/adapters/next-app-dir/nextAppDirCaller.mjs
@@ -2,7 +2,7 @@ import { TRPCError, getTRPCErrorFromUnknown } from '../../unstable-core-do-not-i
 import '../../unstable-core-do-not-import/rootConfig.mjs';
 import { formDataToObject } from './formDataToObject.mjs';
 import { TRPCRedirectError } from './redirect.mjs';
-import { rethrowNextErrors } from './rethrowNextErrors.mjs';
+import { unstable_rethrow} from 'next/navigation.js'
 
 /**
  * Create a caller that works with Next.js React Server Components & Server Actions
@@ -32,7 +32,7 @@ import { rethrowNextErrors } from './rethrowNextErrors.mjs';
                 path,
                 type: opts._def.type
             });
-            rethrowNextErrors(error);
+            unstable_rethrow(error);
             throw error;
         };
         switch(opts._def.type){
diff --git a/dist/adapters/next-app-dir/rethrowNextErrors.d.ts b/dist/adapters/next-app-dir/rethrowNextErrors.d.ts
deleted file mode 100644
index f0e2abd18dfb76b64cf69554330f650a0236c7d2..0000000000000000000000000000000000000000
diff --git a/dist/adapters/next-app-dir/rethrowNextErrors.d.ts.map b/dist/adapters/next-app-dir/rethrowNextErrors.d.ts.map
deleted file mode 100644
index 8eedc88608f6667adcd2329e3c7d7fd82ec8e398..0000000000000000000000000000000000000000
diff --git a/dist/adapters/next-app-dir/rethrowNextErrors.js b/dist/adapters/next-app-dir/rethrowNextErrors.js
deleted file mode 100644
index 8b3cb4227ce9bb847afc25c83155ea228b914ef3..0000000000000000000000000000000000000000
diff --git a/dist/adapters/next-app-dir/rethrowNextErrors.mjs b/dist/adapters/next-app-dir/rethrowNextErrors.mjs
deleted file mode 100644
index 7a702598988fba828d4c339479aed8a2495479a0..0000000000000000000000000000000000000000
