diff --git a/node_modules/next-auth/lib/index.js b/node_modules/next-auth/lib/index.js
index 5a75f39..d123a9a 100644
--- a/node_modules/next-auth/lib/index.js
+++ b/node_modules/next-auth/lib/index.js
@@ -9,17 +9,7 @@ async function getSession(headers, config) {
         headers: { cookie: headers.get("cookie") ?? "" },
     });
     config.useSecureCookies ?? (config.useSecureCookies = origin.protocol === "https:");
-    return Auth(request, {
-        ...config,
-        callbacks: {
-            ...config.callbacks,
-            // Since we are server-side, we don't need to filter out the session data
-            // See https://nextjs.authjs.dev/v5#authenticating-server-side
-            session({ session, user, token }) {
-                return { ...session, user: user ?? token };
-            },
-        },
-    });
+    return Auth(request, config);
 }
 function isReqWrapper(arg) {
     return typeof arg === "function";
