diff --git a/node_modules/next/server.js b/node_modules/next/server.js
index 1fe002a..53325c1 100644
--- a/node_modules/next/server.js
+++ b/node_modules/next/server.js
@@ -3,8 +3,6 @@ const serverExports = {
     .NextRequest,
   NextResponse: require('next/dist/server/web/spec-extension/response')
     .NextResponse,
-  ImageResponse: require('next/dist/server/web/spec-extension/image-response')
-    .ImageResponse,
   userAgentFromString: require('next/dist/server/web/spec-extension/user-agent')
     .userAgentFromString,
   userAgent: require('next/dist/server/web/spec-extension/user-agent')
@@ -23,7 +21,6 @@ module.exports = serverExports
 // make import { xxx } from 'next/server' work
 exports.NextRequest = serverExports.NextRequest
 exports.NextResponse = serverExports.NextResponse
-exports.ImageResponse = serverExports.ImageResponse
 exports.userAgentFromString = serverExports.userAgentFromString
 exports.userAgent = serverExports.userAgent
 exports.URLPattern = serverExports.URLPattern
