diff --git a/server.js b/server.js
index 1fe002aa1333bdc51cdf9cb21cde49c18c0aeb7a..277dd0f07be483dba17e0385129983520c55318e 100644
--- a/server.js
+++ b/server.js
@@ -3,8 +3,8 @@ const serverExports = {
     .NextRequest,
   NextResponse: require('next/dist/server/web/spec-extension/response')
     .NextResponse,
-  ImageResponse: require('next/dist/server/web/spec-extension/image-response')
-    .ImageResponse,
+  // ImageResponse: require('next/dist/server/web/spec-extension/image-response')
+  //   .ImageResponse,
   userAgentFromString: require('next/dist/server/web/spec-extension/user-agent')
     .userAgentFromString,
   userAgent: require('next/dist/server/web/spec-extension/user-agent')
@@ -23,7 +23,7 @@ module.exports = serverExports
 // make import { xxx } from 'next/server' work
 exports.NextRequest = serverExports.NextRequest
 exports.NextResponse = serverExports.NextResponse
-exports.ImageResponse = serverExports.ImageResponse
+// exports.ImageResponse = serverExports.ImageResponse
 exports.userAgentFromString = serverExports.userAgentFromString
 exports.userAgent = serverExports.userAgent
 exports.URLPattern = serverExports.URLPattern